src/enkf/gridio_gfs.f90

```diff
@@ -2876,15 +2880,18 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,grdin,n
         endif
         vg = values_1d + vg*100_r_kind ! analysis ps (values_1d is background ps)
         do k=1,nlevs
-           ug=(rd/grav)*reshape(tv_anal(:,:,nlevs-k+1),(/nlons*nlats/))
+           krev = nlevs-k+1 ! k is bottom to top, krev is top to bottom
```

Comment:
`krev` is not declared in `subroutine writegriddata`.   This causes compilation to fail.